common/utils/retry_test.go

```diff
@@ -0,0 +1,93 @@
+package utils
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRetryTimeout(t *testing.T) {
+	start := time.Now()
+	fn := func() (bool, error) {
+		time.Sleep((time.Duration(rand.Intn(2))) * time.Millisecond)
+		return true, nil
+	}
+
+	if err := Retry(time.Millisecond*1, time.Millisecond*1, fn); err != nil {
+		assert.WithinDuration(t, start.Add(time.Millisecond*1), time.Now(), time.Millisecond)
+	}
+}
+
+func TestRetryNoTimeout(t *testing.T) {
+	retries := 0
+	fn := func() (bool, error) {
+		retries++
+		time.Sleep(1 * time.Millisecond)
+
+		if retries == 5 {
+			return false, nil
+		}
+
+		return true, nil
+	}
+
+	if err := Retry(time.Millisecond*100, time.Millisecond*1, fn); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestRetryFuncTimeRandom(t *testing.T) {
+	start := time.Now()
+	fn := func() (bool, error) {
+		time.Sleep((time.Duration(rand.Intn(2))) * time.Millisecond)
+		return true, nil
+	}
+
+	if err := Retry(time.Millisecond*5, time.Millisecond*1, fn); err != nil {
+		assert.WithinDuration(t, time.Now(), start.Add(time.Millisecond*5), time.Millisecond)
```

Comment:
this doesn't execute if err == nil, and this assertion actually isn't affected at all by the rest of the test